devices/cmos.go

```diff
@@ -0,0 +1,119 @@
+package devices
+
+import (
+	"time"
+)
+
+const (
+	indexMask   = uint8(0x7F)
+	indexOffset = uint64(0x0)
+	dataOffset  = uint64(0x1)
+	dataLen     = uint64(128)
+)
+
+type CMOS struct {
+	Index uint8
+	Data  []uint8
+}
+
+func NewCMOS(memBelow4G, memAbove4G uint64) *CMOS {
+	cmos := &CMOS{
+		Index: 0,
+		Data:  make([]uint8, dataLen),
+	}
+
+	// We assume 3G RAM at all times for now.....
+	extMem := uint16(0xBC00)
+
+	cmos.Data[0x34] = uint8(extMem)
+	cmos.Data[0x35] = uint8(extMem >> 8)
+
+	// Only valid for PVH boot of firmware with 3G RAM fixed.....
+	cmos.Data[0x5b] = 0
+	cmos.Data[0x5c] = 0
+	cmos.Data[0x5d] = 0
+
+	return cmos
+}
+
+func (c *CMOS) Read(base uint64, data []byte) error {
+	if len(data) != 1 {
+		return errDataLenInvalid
+	}
+
+	var d uint8
+
+	switch base ^ 0x70 {
```

Comment:
Is there a datasheet on the offset and index used within this switch block?
If so, it would be greatly helpful if you could leave a comment.